uaclient/contract.py

```diff
@@ -636,31 +639,32 @@ def is_contract_changed(cfg: UAConfig) -> bool:
         .get("contractInfo", {})
         .get("id", "")
     )
-    contract_client = UAContractClient(cfg)
-    resp = contract_client.get_updated_contract_info(
-        machine_token, contract_id
-    )
-    resp_expiry = (
-        resp.get("machineTokenInfo", {})
-        .get("contractInfo", {})
-        .get("effectiveTo", None)
-    )
-    new_expiry = (
-        util.parse_rfc3339_date(resp_expiry)
-        if resp_expiry
-        else cfg.machine_token_file.contract_expiry_datetime
-    )
-    if cfg.machine_token_file.contract_expiry_datetime != new_expiry:
-        return True
-    curr_entitlements = cfg.machine_token_file.get_entitlements_from_token(
-        resp
-    )
-    for name, new_entitlement in sorted(curr_entitlements.items()):
-        deltas = util.get_dict_deltas(
-            orig_entitlements.get(name, {}), new_entitlement
+    if contract_id != "":
```

Comment:
I think it is simpler to just return False here if the `contract_id` is None. We can check for None if we modify get to not have an empty string as a default value when we don't have the `id` key